packages/txm/lib/TxMonitor.ts

```diff
@@ -372,6 +372,17 @@ export class TxMonitor {
             return transaction.changeStatus(TransactionStatus.Expired)
         }
 
+        if (transaction.collectionBlock && transaction.collectionBlock > block.number - 5n) {
+            // Skip transactions collected in the last 5 blocks to prevent race conditions between
+            // the transaction collector and monitor processing the same transaction simultaneously
+            span.addEvent("txm.tx-monitor.handle-not-attempted-transaction.skip-transaction", {
+                transactionIntentId: transaction.intentId,
+                collectionBlock: Number(transaction.collectionBlock),
+                blockNumber: Number(block.number),
+            })
+            return
+        }
```

Comment:
Waiting 5 blocks might introduce a lot of lag in our retries.
I'm retooling this so that we mark transaction that haven't gone through the initial submission as a new `TransactionStatus.NotAttempted` and we filter to not process these transactions in the `TxMonitor`.